.github/workflows/post_publish.yml

```diff
@@ -51,3 +51,84 @@ jobs:
     if: ${{ github.event.workflow_run.conclusion == 'failure' }}
     steps:
       - run: echo 'The triggering workflow failed'
+
+  registry-check:
+    runs-on: ubuntu-latest
+    needs: [on-success-or-workflow-dispatch]
+    outputs:
+      latest-version: ${{ steps.registry_latest_ver.outputs.current }}
+    steps:
+      - name: Registry Version Check
+        id: registry_latest_ver
+        shell: bash
+        run: |
+          for i in 1 2
+          do
+            LATEST_VERSION=$(curl -s "https://registry.terraform.io/v2/providers/323/provider-versions/latest" | jq -r '.data.attributes.version')
+            if [[ " ${{ needs.on-success-or-workflow-dispatch.outputs.release-tag }}" != ${LATEST_VERSION} ]]; then
+              sleep 1h
+            else
+              echo "Registry and Github Version matches"
+                echo "current=$LATEST_VERSION >> "$GITHUB_OUTPUT"
+            fi
+          done
+
+  os-version-init:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    runs-on: ${{ matrix.os }}
+    needs: [registry-check]
+    steps:
+      - name: Run Terraform Init on the os-version-init
+        run: |
+          echo "Setting provider version in Terraform configuration..."
+          cat <<EOF > main.tf
+          terraform {
+            required_providers {
+              aws = {
+                source  = "hashicorp/aws"
+                version = ${{ needs.registry-check.outputs.latest-version }}
+              }
+            }
+          }
+
+          provider "aws" {
+            region = "us-east-1"
+          }
+          EOF
+
+          echo "Terraform configuration file (main.tf) created with the latest version."
+
+          echo "Initializing Terraform with the latest AWS provider..."
+          terraform init -upgrade
+
+          if [ $? -ne 0 ]; then
+            echo "Terraform initialization failed."
+            exit 1
+          fi
```

Comment:
I believe we can eliminate this part of the script with the above comment in mind, since the step would fail if `terraform init -upgrade` fails.